### PR TITLE
dpdk drop 2

### PR DESCRIPTION
**Pull request by ahalimx86**
Signed-off-by: Abdul Halim <abdul.halim@intel.com>

Co-Authored-By: nupurjai <nupurjai@users.noreply.github.com>
Co-Authored-By: lpawlikx <lpawlikx@users.noreply.github.com>
Co-Authored-By: ipatrykx <ipatrykx@users.noreply.github.com>
Co-Authored-By: vishala-r <vishala-r@users.noreply.github.com>
Co-Authored-By: dhruvpt <dhruvpt@users.noreply.github.com>
Co-Authored-By: venkateshpnv <venkateshpnv@users.noreply.github.com>
Co-Authored-By: shweta503 <shweta503@users.noreply.github.com>
Co-Authored-By: Eoghan1232 <Eoghan1232@users.noreply.github.com>
Co-Authored-By: eoghanlawless <eoghanlawless@users.noreply.github.com>
Co-Authored-By: mgaravin <mgaravin@users.noreply.github.com>

### CODE REVIEW

**Review by nupurjai — APPROVED** [review]

**Review by dhruvpt — APPROVED** [review]

**Comment by nupurjai** [comment]
looks ok to me.

**Comment by dhruvpt** [comment]
Did combined review --- looks good!

File: inframanager/api_handler/api_handler.go [hunk 1]
```diff
@@ -197,41 +258,27 @@ func (s *ApiServer) CreateNetwork(ctx context.Context, in *proto.CreateNetworkRe
...
-		logger.Errorf("Failed to insert the entries for %s %s", macAddr, ipAddr)
+	portID, err := strconv.ParseInt(portIDStr, 10, 32)
+	if err != nil {
+		logger.Errorf("Failed to convert port id to int %s", portIDStr)
 		out.Successful = false
 		return out, err
 	}
 
-	logger.Infof("Inserted the entries %s %s", macAddr, ipAddr)
+	status, err := insertRule(s.log, ctx, server.p4RtC, macAddr,
```
Comment (vishala-r) [inline inframanager/api_handler/api_handler.go:275]: in delete case, we have mentioned DeleteCniRules, i think the same naming convention should be used here as well, insertCniRules()

File: inframanager/api_handler/api_handler.go [hunk 2]
```diff
@@ -243,27 +290,34 @@ func (s *ApiServer) DeleteNetwork(ctx context.Context, in *proto.DeleteNetworkRe
...
-	if err != nil {
-		logger.Errorf("Error when getting forwarding pipe: %v", err)
+	ipAddr := strings.Split(in.Ipv4Addr, "/")[0]
+	macAddr := in.MacAddr
+
+	ep := store.EndPoint{
+		PodIpAddress: ipAddr,
+	}
+
+	entry := ep.GetFromStore()
```
Comment (vishala-r) [inline inframanager/api_handler/api_handler.go:300]: we can take this operation inside DeleteCniRules(), since we are doing the same in case of InsertRule()

File: inframanager/api_handler/api_handler.go [hunk 3]
```diff
@@ -452,6 +506,38 @@ func (s *ApiServer) UpdateGlobalBGPConfig(ctx context.Context, in *proto.GlobalB
...
+	logger := s.log.WithField("func", "SetupHostInterface")
+	logger.Infof("Incoming SetupHostInterface reqest %s", in.String())
+
+	out := &proto.Reply{
+		Successful: true,
+	}
+
+	server := NewApiServer()
+
+	ipAddr := strings.Split(in.Ipv4Addr, "/")[0]
```
Comment (vishala-r) [inline inframanager/api_handler/api_handler.go:521]: may be a common function to do split of the input string we received and return the array of values would be good. it can be taken in another commit also.

**Review by vishala-r — APPROVED** [review]

**Comment by vishala-r** [comment]
looks good to me. the submitted feedback can be taken in different commit too.

File: inframanager/api_handler/api_handler.go (hunk 3 above)
Comment (ahalimx86) [reply inframanager/api_handler/api_handler.go:521]: > looks good to me. the submitted feedback can be taken in different commit too.

Agreed. I would prefer not to make changes to others work on this PR.
